src/main/java/org/sagebionetworks/bridge/udd/exceptions/AsyncTaskExecutionException.java

```diff
@@ -0,0 +1,24 @@
+package org.sagebionetworks.bridge.udd.exceptions;
+
+/**
+ * This represents an exception in an asynchronous Runnable. This is a RuntimeException because Runnable doesn't
+ * declare any checked exceptions. This is generally converted into an ExecutionException by the ExecutorService and
+ * Future.get().
```

Comment:
Good catch. These were originally Runnables, but I replaced them with Callables at some point. I failed to realize that Callable throws Exception. I've gone ahead and made this a checked Exception instead of a RuntimeException and simplified a few places in the code.

That said, there are some places where we use AsyncTaskExecutionException to enhance the error message (see https://github.com/DwayneJengSage/BridgeUserDataDownloadService/blob/develop/src/main/java/org/sagebionetworks/bridge/udd/synapse/SynapseDownloadFromTableTask.java#L133), so we'll still need this exception.